src/main/groovy/io/sentry/android/gradle/SentryPlugin.groovy

```diff
@@ -274,60 +243,48 @@ class SentryPlugin implements Plugin<Project> {
 
                     def cli = getSentryCli(project)
 
-                    def persistIdsTaskName = "persistSentryProguardUuidsFor${variant.name.capitalize()}${variantOutput.name.capitalize()}"
-                    // create a task that persists our proguard uuid as android asset
-                    def persistIdsTask = project.tasks.create(
-                            name: persistIdsTaskName,
-                            type: Exec) {
-                        description "Write references to proguard UUIDs to the android assets."
-                        workingDir project.rootDir
-
-                        def propsFile = getPropsString(project, variant)
+                    def generateUuidTask = project.tasks.create(
+                            name: "generateSentryProguardUuid${variant.name.capitalize()}${variantOutput.name.capitalize()}",
+                            type: GenerateSentryProguardUuidTask) {
+                        outputDirectory.set(project.file("build/generated/assets/sentry/${variant.name}"))
 
-                        if (propsFile != null) {
-                            environment("SENTRY_PROPERTIES", propsFile)
-                        } else {
-                            project.logger.info("propsFile is null")
+                        doFirst {
+                            project.logger.info("debugMetaPropPath: ${getOutputFile().get()}")
                         }
+                    }
 
-                        def debugMetaPropPath = getDebugMetaPropPath(project, variant)
-                        project.logger.info("debugMetaPropPath: ${debugMetaPropPath}")
-
-                        def args = [
-                                cli,
-                                "upload-proguard",
-                                "--write-properties",
-                                debugMetaPropPath,
-                                mappingFile
-                        ]
-
-                        if (!extension.autoUpload) {
-                            args << "--no-upload"
+                    if (VersionNumber.parse(Version.ANDROID_GRADLE_PLUGIN_VERSION) >= new VersionNumber(3, 3, 0, null)) {
```

Comment:
this plugin should work across multiple Gradle and AGP versions, not sure if with the new changes, this still holds true, as the bug that this PR tries to solve starts happening on Gradle >= 6.7 or AGP >= 4.0, it'd be fine to bump a minor version and do the breaking change, people still could use older versions if they don't have the newest AGP/Gradle versions, but we'd need to be testing this fix across Gradle >= 6.7 and AGP >= 4.0, AGP probably up to 4.2, as 7.0 is very alpha.

Reply:
Yeah, I agree. In general it feels better to support a smaller set of versions and just let people use the old versions. But for this change I did not want to break anything, though AGP 3.3 is pretty damn old.